components/login/provider/index.jsx

```diff
@@ -33,217 +33,203 @@ import {
       <div className={bem("provider-login__wrapper")}>
         <FormControl
-          classes={{ root: classes.selectionBox }}
+          classes={{ root: bem("provider-login__formcontrol") }}
           error={!!loginError}
         >
           <Autocomplete
-            onChange={onProviderChange}
+            onChange={handleProviderChange}
+            onInputChange={handleInputChange}
             id="provider-select"
             freeSolo
-            options={providersWithIdp}
+            options={defaultIdentityProviders}
             getOptionLabel={(option) => option.label ?? option}
-            value={provider?.iri || providerIri}
+            defaultValue={provider?.iri}
             autoSelect
             renderOption={(option) => {
               return (
                 <>
-                  {option.logo ? (
+                  {option.logo && (
                     <img
                       src={`../${option.logo}`}
                       width={20}
-                      style={{ margin: "1rem" }}
                       alt={option.label}
                     />
-                  ) : (
-                    <div style={{ minWidth: "20px", padding: "1.6rem" }} />
                   )}
-                  {option.label}
+                  <div
+                    style={{
+                      marginLeft: option.logo ? "0px" : "20px",
+                      padding: "0.5rem 1rem",
+                    }}
+                  >
+                    {option.label}
+                  </div>
                 </>
               );
             }}
             renderInput={(params) => {
               return (
                 <TextField
-                  onChange={handleChange}
                   {...params}
+                  inputProps={{
+                    ...params.inputProps,
+                    autoComplete: "off",
+                    inputMode: "url",
+                  }}
                   error={!!loginError}
                   margin="none"
                   variant="outlined"
-                  type="url"
                   aria-describedby={loginError ? "login-error-text" : null}
                   data-testid={TESTCAFE_ID_LOGIN_FIELD}
                   ref={loginFieldRef}
                   value={inputValue}
-                  onKeyUp={(e) => {
-                    if (e.key === "Enter") {
-                      handleLogin(e, providerIri);
-                    }
-                  }}
                 />
               );
             }}
           />
           {loginError ? (
-            <FormHelperText
-              id="login-error-text"
-              style={{ marginBottom: theme.spacing(1) }}
-            >
-              {getErrorMessage(loginError)}
+            <FormHelperText id="login-error-text" style={{ marginBottom: 0 }}>
+              {loginError}
             </FormHelperText>
-          ) : null}
+          ) : (
+            <p style={{ margin: 0 }}>&nbsp;</p>
```

Comment:
Small little hack to make the page not "jump" when the single-line error message is shown.